### PR TITLE
Basic tests

### LINKED ISSUE #13: Tests required
There should be provided some tests that check lib's functionality. Tests should be run in CI by pytest.

### PR DESCRIPTION
**Pull request by ivigns**
Fixes #13

### CODE REVIEW

**Comment by ivigns** [comment]
More tests will be added till this Thursday's evening.

File: tests/test_encvector.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from pysdsl import *
+
+
+def test_encvector():
+    for T in [EncVectorEliasDelta, EncVectorEliasGamma, EncVectorFibonacci,
+              EncVectorComma4, EncVectorComma2]:
```
Comment (code-of-kpp) [inline tests/test_encvector.py]: please parametrize them with pytest.mark.parametrize and use `all_*` lists from package to test that

File: .droneci/config.yml [hunk 2]
```diff
@@ -53,12 +53,11 @@ steps:
...
-  - apt-get update -qq
-  - apt-get install -qqy python python-pip
+  - apt-get -qq update
+  - apt-get -qqy install python python-pip
   - pip install $(ls -d /tmp/cache/dist/*)
-  - mkdir /tmp/test
-  - cd /tmp/test
-  - python -c 'import pysdsl'
+  - pip install -U pytest
+  - pytest tests
```
Comment (code-of-kpp) [inline .droneci/config.yml]: please run pytest with  `--verbose --showlocals --color=yes` (you probably don't need to specify tests folder)

File: tests/test_encvector.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from pysdsl import *
```
Comment (code-of-kpp) [inline tests/test_encvector.py]: please just `import pytest` don't use `*` import

File: tests/test_suffixarray.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from pysdsl import *
+
+
+def test_suffixarray():
+    for T in [SuffixArraySadakaneInt, SuffixArrayWaveletTreeInt,
+              SuffixArrayWaveletTree, SuffixArraySadakane,
+              SuffixArrayBitcompressed]:
```
Comment (code-of-kpp) [inline tests/test_suffixarray.py]: you don't really use T

File: tests/test_wavelet.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from pysdsl import *
+
+
+def test_wavelet():
+    for T in all_wavelet_trees:
+        a = T([3, 2, 1, 0, 2, 1, 3, 4, 1, 1, 1, 3, 2, 3])
+        a.size_in_bytes
```
Comment (code-of-kpp) [inline tests/test_wavelet.py]: there is no asserts and/or expected Exceptions here

**Review by code-of-kpp — CHANGES_REQUESTED** [review]

**Comment by ivigns** [comment]
Finishing fixes today

File: tests/test_encvector.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import pysdsl
+import pytest
+
+
+@pytest.mark.parametrize("Type", pysdsl.all_compressed_integer_vectors)
+def test_encvector(Type):
+    if Type == pysdsl.pysdsl.DirectAccessibleCodesVectorDP or \
```
Comment (code-of-kpp) [inline tests/test_encvector.py]: is it to just pass the tests for now? It is better then to split the test and mark the broken one with pytest.mark.skip

File: tests/test_intvector.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import pysdsl
+import pytest
+
+
+def test_intvector():
+    v = pysdsl.IntVector(1024 * 1024 * 256)
+    assert v.size_in_mega_bytes < 2049
```
Comment (code-of-kpp) [inline tests/test_intvector.py]: why isn't it exact comparison?

File: tests/test_wavelet.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import pysdsl
+import pytest
+
+
+@pytest.mark.parametrize("Type", pysdsl.all_wavelet_trees)
+def test_wavelet(Type):
+    a = Type([3, 2, 1, 0, 2, 1, 3, 4, 1, 1, 1, 3, 2, 3])
+    assert a.select(2, 3) == 6 or a.select(2, 3) == 49  # 2nd is for Huffman
```
Comment (code-of-kpp) [inline tests/test_wavelet.py]: why is that 49? Maybe we need a separate test here?

File: tests/test_encvector.py (hunk 6 above)
Comment (code-of-kpp) [reply tests/test_encvector.py]: Also please link here corresponding issue.

**Review by code-of-kpp — CHANGES_REQUESTED** [review]

File: tests/test_encvector.py (hunk 6 above)
Comment (ivigns) [reply tests/test_encvector.py]: Here's the corresponding issue #16

**Comment by ivigns** [comment]
Fixing commentaries tomorrow asap

**Pull request merged**